File: Tyche.Android/pool/src/main/java/com/felipearpa/tyche/pool/PoolGamblerScoreTransformer.kt
package com.felipearpa.tyche.pool

import com.felipearpa.tyche.data.pool.domain.PoolGamblerScore

fun PoolGamblerScore.toPoolGamblerScoreModel() =
    PoolGamblerScoreModel(
        poolId = this.poolId,
        poolName = this.poolName,
        gamblerId = this.gamblerId,
        gamblerUsername = this.gamblerUsername,
        currentPosition = this.currentPosition,
        beforePosition = this.beforePosition,
        score = this.score
    )

File: Amazon.Lambda/Felipearpa.Tyche.Function/src/Response/PoolGamblerScoreTransformer.fs
namespace Felipearpa.Tyche.Function.Response

open Felipearpa.Tyche.Pool.Domain
open Felipearpa.Type

module PoolGamblerScoreTransformer =

    let toResponse (poolGamblerScore: PoolGamblerScore) =
        { PoolGamblerScoreResponse.PoolId = poolGamblerScore.PoolId |> Ulid.value
          PoolName = poolGamblerScore.PoolName |> NonEmptyString100.value
          GamblerId = poolGamblerScore.GamblerId |> Ulid.value
          GamblerUsername = poolGamblerScore.GamblerUsername |> NonEmptyString100.value
          Position = poolGamblerScore.Position
          BeforePosition = poolGamblerScore.BeforePosition
          Score = poolGamblerScore.Score
          BeforeScore = poolGamblerScore.BeforeScore }
